vm.c

```diff
@@ -137,19 +128,21 @@ static inline vm_value *vm_get_op_value(vm_env *env, const vm_operand *op)
 void vm_run(vm_env *env)
 {
     size_t pc = 0;
-    BEGIN_OPCODES;
-
-    OP(ADD) : VM_CALL_HANDLER();
-    DISPATCH;
-    OP(SUB) : VM_CALL_HANDLER();
-    DISPATCH;
-    OP(PRINT) : VM_CALL_HANDLER();
-    DISPATCH;
-    OP(JMP) : VM_GOTO(OPCODE.op1.value.id);
-
-    OP(HALT) : goto terminate;
+    for (;;) {
+next_opcode:
```

Comment:
I see. 

Add a additional reference about this: http://www.cs.toronto.edu/~matz/dissertation/matzDissertation-latex2html/node6.html